armi/reactor/converters/geometryConverters.py

```diff
@@ -96,10 +96,6 @@ def reset(self):
         runLog.info(
             f"Resetting the state of the converted reactor core model in {self}"
         )
-        currentAssemCounter = assemblies.getAssemNum()
-        assemblies.setAssemNumCounter(
-            currentAssemCounter - len(self._newAssembliesAdded)
-        )
```

Comment:
I understand that the `setAssemNumCounter()` machinery has been removed, so obviously it can no longer be called.

But don't we still need to fix the assembly number counter parameter on the reactor when moving from one core representation to another?

Reply:
Well, I didn't / don't think so.

I believe the chain of logic goes like this:

1. The original core gets built.
2. At this point, the `maxAssemNum` is whatever... above zero, Let's say "N".
3. Some converter (geometryConvert") is called
4. If the converter doesn't add new assemblies, we're fine: `maxAssemNum == N`
5. If the converter DOES add new assemblies (which is common), the new assembly numbers will start with `N` and go up.
6. The end result will be `maxAssemNum == M` where `M > N`.

I believe that situation is sufficient for our needs, correct, and the same as we have now.

BUT

If you think I'm just seeing the problem, speak up.  If I've missed something, I'm happy to improve the logic here.

Reply:
I agree with your chain of logic steps.

But what about when the reactor gets converted back down? So maybe it started as a 1/3 core representation, we convert to full-core to do some CR worth calcs, and then bring it back down to 1/3 core to continue the remainder of the calculation sequence.

In that case, my expectation would be that `r.maxAssemNum` would be correspond to the 1/3 core representation for the remainder of the calculation sequence.

But that could be a fundamental misunderstanding of mine. I think both ways are legitimate, but I guess I don't know which one is "correct" for us.

I suppose in the past `maxAssemNum` was always monotonically increasing and it would include any temporary core conversions. Based on that, I guess that your current logic is correct.

Reply:
ARMI does not currently preserve the assembly numbers when you go from 1/3 core to full core and back again.

In order to achieve that, we would need to do a lot of bookkeeping, At the very least we would need to keep a record of what conversions have been done.  And store a bunch of key information from the Reactor state before the conversion. Which we don't do.

Also, I imagine reverting a converter would be impossible after one time step of modeling takes place.